recipe/quadrature/quadrature.cc

```diff
@@ -0,0 +1,39 @@
+#include "recipe/quadrature/quadrature.h"
+#include <cassert>
+
+namespace recipe {
+namespace quadrature {
+
+double TrapezoidalRule(double (*f)(double), const double a, const double b,
```

Comment:
It is safe to put assert for a case of `num_of_partition <= 0` or add a comment to documents of this function.